src/components/ToggleCollapse.tsx

```diff
@@ -0,0 +1,188 @@
+import { CaretDownFilled, CaretUpFilled } from "@ant-design/icons";
```

Comment:
![Image](https://github.com/user-attachments/assets/004e738b-98f9-4051-82a8-cbd0418a094c)

Reply:
For this toggle switch and collapse arrow in the same section, you explained in the walkthrough how they work together, and I know they've been through a few ux reviews, but i'm still not that clear on why we need both when they overlap in hiding the settings. Should they be combined into one control or kept separate completely with distinct purposes? I'd like to learn more about the reasoning i might not be aware.